src/Xdl.Internship.Core/Xdl.Internship.Core.DataAccess/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/IMongoRepository.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xdl.Internship.Core.DataAccess.MongoDB.Entities;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.Repositories
+{
+    public interface IMongoRepository<TDocument> where TDocument : IDocument
```

Comment:
MongoDriver allow to use CancellationToken, should be added to our repo methods too